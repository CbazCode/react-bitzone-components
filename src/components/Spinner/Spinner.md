```js
<Spinner/>
```
```js
<Spinner size="sm" color="warning"/>
```
```js
<Spinner size="md" color="danger"/>     
```
```js
<Spinner size="lg" color="primary"/>
```