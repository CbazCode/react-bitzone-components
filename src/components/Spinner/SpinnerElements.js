import styled, { keyframes } from 'styled-components';
import { setBgColor } from '../../utils/helpers';


//@atributes
// color = "red" - "yellow" - "green" - "blue" - "purple" - "indigo" - "pink" - default
// size = "sm" - "md" - "lg"

const spinnerOne = keyframes`
    0% { transform: rotate(0deg); border-width: 8px; }
    50% { transform: rotate(180deg); border-width: 1px; }
    100% { transform: rotate(360deg); border-width: 8px; }
`;

const spinnerTwo = keyframes`
    0% { transform: rotate(0deg); border-width: 1px; }
    50% { transform: rotate(180deg); border-width: 8px; }
    100% { transform: rotate(360deg); border-width: 1px; }
`;


const setSize = (size) => {
    switch (size) {
        case "sm":
            return '50px';
        case "md":
            return '100px';
        case "lg":
            return '200px';
        default:
            return '100px';
    }
}



export const SpinnerBody = styled.div` 
    width: ${({ size }) => setSize(size)};
    height: ${({ size }) => setSize(size)};
    position: relative;
    margin: 0.5rem
`

export const SpinnerPart = styled.div`
    box-sizing: border-box;
    position: absolute;
    width: 100%;
    height: 100%;
    border: 8px solid transparent;
    border-top-color: ${({ color }) => setBgColor(color)};
    border-radius: 50%;
    animation: ${spinnerOne} 1.2s linear infinite;

    &:nth-child(2){
        border: 8px solid transparent;
        border-bottom-color: ${({ color }) => setBgColor(color)};
        animation: ${spinnerTwo} 1.2s linear infinite;
    }
`
