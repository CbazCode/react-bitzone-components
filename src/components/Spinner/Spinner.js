import React from 'react'
import { SpinnerBody, SpinnerPart } from './SpinnerElements'

const Spinner = ({size, color}) => {
    return (
        <SpinnerBody size={size}>
            <SpinnerPart color={color}/>
            <SpinnerPart color={color}/>
        </SpinnerBody>
    )
}

export default Spinner;
