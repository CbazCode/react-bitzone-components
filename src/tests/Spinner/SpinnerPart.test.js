import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import { SpinnerPart } from '../../components/Spinner/SpinnerElements';

describe('Pruebas en compnente <SpinnerPart/>', () => {
    test('debe renderizarse correctamente', () => {
        const { container } = render(<SpinnerPart />);
        expect(container.firstChild).toMatchSnapshot();
    })

    test('debe renderizarse con estilo por defecto', () => {
        const spinner = renderer.create(<SpinnerPart/>).toJSON();
        expect(spinner).toHaveStyleRule('width', '100%');
        expect(spinner).toHaveStyleRule('height', '100%');
        expect(spinner).toHaveStyleRule('position', 'absolute');
        expect(spinner).toHaveStyleRule('border', '8px solid transparent');
        expect(spinner).toHaveStyleRule('border-top-color', '#9CA3AF');
        expect(spinner).toHaveStyleRule('border-radius', '50%');
    })

    test('debe renderizarse con el tamaÃ±o escogido por el usuario', () => {
        const spinner = renderer.create(<SpinnerPart color="blue" />).toJSON();
        expect(spinner).toHaveStyleRule('width', '100%');
        expect(spinner).toHaveStyleRule('height', '100%');
        expect(spinner).toHaveStyleRule('position', 'absolute');
        expect(spinner).toHaveStyleRule('border', '8px solid transparent');
        expect(spinner).toHaveStyleRule('border-top-color', '#3B82F6');
        expect(spinner).toHaveStyleRule('border-radius', '50%');
    })
   
    
})