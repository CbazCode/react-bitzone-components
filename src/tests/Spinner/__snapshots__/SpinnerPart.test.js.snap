// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en compnente <SpinnerPart/> debe renderizarse correctamente 1`] = `
.c0 {
  box-sizing: border-box;
  position: absolute;
  width: 100%;
  height: 100%;
  border: 8px solid transparent;
  border-top-color: #9CA3AF;
  border-radius: 50%;
  -webkit-animation: bwOoWs 1.2s linear infinite;
  animation: bwOoWs 1.2s linear infinite;
}

.c0:nth-child(2) {
  border: 8px solid transparent;
  border-bottom-color: #9CA3AF;
  -webkit-animation: dotDJp 1.2s linear infinite;
  animation: dotDJp 1.2s linear infinite;
}

<div
  class="c0"
/>
`;
